spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -281,6 +281,12 @@ public boolean canDeleteWhere(Filter[] filters) {
   private boolean canDeleteUsingMetadata(Expression deleteExpr) {
     boolean caseSensitive =
         Boolean.parseBoolean(sparkSession().conf().get("spark.sql.caseSensitive"));
+
+    String branch =
+        sparkSession().conf().contains(SparkWriteOptions.BRANCH)
+            ? sparkSession().conf().get(SparkWriteOptions.BRANCH)
+            : null;
```

Comment:
This is another area that should not use a `SparkWriteOption` with session conf.

Reply:
https://github.com/apache/iceberg/pull/6651#discussion_r1103847386